File: src/sub/accunormeqs.c
#include <stdlib.h>

void outer(double *g, const double *rowofa, int nvar, const double *w);
void rside(double *rhs, const double *rowofa, int nvar, const double *res, const double *w);

void accunormeqs(double *rowofa, int nvar, double res, double w,
                 double *g, double *rhs) {
    outer(g, rowofa, nvar, &w);
    rside(rhs, rowofa, nvar, &res, &w);
}

File: src_fortran/sub/accunormeqs.f
c
c
c
c
c
c
      subroutine ACCUNORMEQS(rowofa,nvar,res,w,g,rhs)
c
c     Urs Kradolfer, 22. 4. 1987
c
c     accumulate normal equations, knowing only one row of A at a time;
c     subr. ACCUNORMEQS must be called nvar times (once for
c     each observation).
c     w is the (normalized ! ) weight of the observation.
c
c        A * x = res          INPUT: one row of A and one element of res
c
c     At*A * x = At*res
c
c       G  * x =  RHS         OUTPUT: G (nvar x nvar) and RHS (nvar)
c
      implicit none
      real g(*),rowofa(*),rhs(*),res,w
      integer nvar
      call OUTER(g,rowofa,nvar,w)
      call RSIDE(rhs,rowofa,nvar,res,w)
c
      return
      end ! of subr. accunormeqs
